src/ng-select/lib/ng-select.component.ts

```diff
@@ -291,27 +286,28 @@ export class NgSelectComponent implements OnDestroy, OnChanges, OnInit, AfterVie
 
 		const term = this.searchTerm.toLowerCase().trim();
 		return (
-			this.addTag &&
+			this.addTag() &&
 			!this.itemsList.filteredItems.some((x) => x.label.toLowerCase() === term) &&
-			((!this.hideSelected && this.isOpen) || !this.selectedItems.some((x) => x.label.toLowerCase() === term)) &&
-			!this.loading
+			((!this.hideSelected() && this.isOpen()) || !this.selectedItems.some((x) => x.label.toLowerCase() === term)) &&
+			!this.loading()
 		);
 	}
 
 	private get _editableSearchTerm(): boolean {
-		return this.editableSearchTerm && !this.multiple;
+		return this.editableSearchTerm() && !this.multiple();
 	}
 
 	private get _isTypeahead() {
-		return this.typeahead && this.typeahead.observed;
+		const typeahead = this.typeahead();
```

Comment:
Nit: I don't we need separate variable here

Reply:
Updated - https://github.com/ng-select/ng-select/pull/2584/commits/53cf59e5182d0cbc93ac9608a82ee5b0dc0c44be